common/buildcraft/lib/block/VanillaRotationHandlers.java

```diff
@@ -62,13 +80,27 @@
+        CustomRotationHelper.INSTANCE.registerHandlerForAll(BlockTripWireHook.class, VanillaRotationHandlers::rotateTripWireHook);
         CustomRotationHelper.INSTANCE.registerHandlerForAll(BlockDoor.class, VanillaRotationHandlers::rotateDoor);
         CustomRotationHelper.INSTANCE.registerHandlerForAll(BlockPistonBase.class, VanillaRotationHandlers::rotatePiston);
         CustomRotationHelper.INSTANCE.registerHandlerForAll(BlockLever.class, VanillaRotationHandlers::rotateLever);
-        CustomRotationHelper.INSTANCE.registerHandlerForAll(BlockFenceGate.class, VanillaRotationHandlers::rotateGate);
-        CustomRotationHelper.INSTANCE.registerHandlerForAll(BlockRedstoneDiode.class, VanillaRotationHandlers::rotateDiode);
+        CustomRotationHelper.INSTANCE.registerHandlerForAll(BlockShulkerBox.class, VanillaRotationHandlers::rotateShulkerBox);
+        CustomRotationHelper.INSTANCE.registerHandlerForAll(BlockDispenser.class, getHandlerFreely(BlockDispenser.class));
+        CustomRotationHelper.INSTANCE.registerHandlerForAll(BlockObserver.class, getHandlerFreely(BlockObserver.class));
+        CustomRotationHelper.INSTANCE.registerHandlerForAll(BlockEndRod.class, getHandlerFreely(BlockEndRod.class));
+        CustomRotationHelper.INSTANCE.registerHandlerForAll(BlockFenceGate.class, getHandlerHorizontalFreely(BlockFenceGate.class));
+        CustomRotationHelper.INSTANCE.registerHandlerForAll(BlockRedstoneDiode.class, getHandlerHorizontalFreely(BlockRedstoneDiode.class));
+        CustomRotationHelper.INSTANCE.registerHandlerForAll(BlockPumpkin.class, getHandlerHorizontalFreely(BlockPumpkin.class));
+        CustomRotationHelper.INSTANCE.registerHandlerForAll(BlockGlazedTerracotta.class, getHandlerHorizontalFreely(BlockGlazedTerracotta.class));
+        CustomRotationHelper.INSTANCE.registerHandlerForAll(BlockAnvil.class, getHandlerHorizontalFreely(BlockAnvil.class));
+        CustomRotationHelper.INSTANCE.registerHandlerForAll(BlockEnderChest.class, getHandlerHorizontalFreely(BlockEnderChest.class));
+        CustomRotationHelper.INSTANCE.registerHandlerForAll(BlockFurnace.class, getHandlerHorizontalFreely(BlockFurnace.class));
+        CustomRotationHelper.INSTANCE.registerHandlerForAll(BlockCocoa.class, VanillaRotationHandlers::rotateCocoa);
         CustomRotationHelper.INSTANCE.registerHandlerForAll(BlockTorch.class, VanillaRotationHandlers::rotateTorch);
+        CustomRotationHelper.INSTANCE.registerHandlerForAll(BlockLadder.class, VanillaRotationHandlers::rotateLadder);
+        CustomRotationHelper.INSTANCE.registerHandlerForAll(BlockHopper.class, VanillaRotationHandlers::rotateHopper);
         CustomRotationHelper.INSTANCE.registerHandlerForAll(BlockChest.class, VanillaRotationHandlers::rotateChest);
+        // TODO: DoubleChest, BannerHanging, BannerStanding, Bed, DoublePlant, EndPortalFrame (?), Stairs, TrapDoor, WallSign, StandingSign, Skull, CommandBlock (?)
```

Comment:
Just leave a comment if I should add/remove something from this todo

Reply:
| BlockType | Coment |
|--|--|
| double chests | Hopefully this can just be alternated between the two directions, although I'm not sure if it needs anything else updating. |
| BlockBed | I'm not sure about beds, as respawning mechanics might require a specific part of the bed. |
| BlockEndPortalFrame | Should probably not be rotatable, as player's can't even break it in survival. Although it might be useful in creative? |
| BlockWallSign, BlockBannerHanging | Can you do the same thing with these as with ladders? (Although they might need different checks than ladders to match vanilla) | 
| BlockStandingSign, BlockBannerStanding  | In theory it should be possible to just increase the rotation angle by one on every rotation. |
| BlockStairs | I think it might be best to cycle round the facing property until we get to a predetermined one (say NORTH), then invert the `EnumHalf` bit so it flips from the top to the bottom instead. (And then it would have to flip on the previous one for bottom to top) |
| BlockTrapDoor | As vanilla doesn't doesn't care if a trap door doesn't face a particular side, we can probably do a similar thing to stairs. |
| BlockSkull | This is a bit trickier as the rotation data is either in the tile entity or in the block state. However it should be possible to do something with the skull - although perhaps treat the rotation types separately - never go from a wall-based skull to a ground-based one. |
| BlockCommandBlock | This might be useful, I'm not sure. This one is a bit sensitive due to command blocks being unforgiving to minor changes (it's basically code). |

Reply:
| BlockType | Coment |
|--|--|
| double chests | Hopefully this can just be alternated between the two directions, although I'm not sure if it needs anything else updating. |
| BlockBed | I'm not sure about beds, as respawning mechanics might require a specific part of the bed. |
| BlockEndPortalFrame | Should probably not be rotatable, as player's can't even break it in survival. Although it might be useful in creative? |
| BlockWallSign, BlockBannerHanging | Can you do the same thing with these as with ladders? (Although they might need different checks than ladders to match vanilla) | 
| BlockStandingSign, BlockBannerStanding  | In theory it should be possible to just increase the rotation angle by one on every rotation. |
| BlockStairs | I think it might be best to cycle round the facing property until we get to a predetermined one (say NORTH), then invert the `EnumHalf` bit so it flips from the top to the bottom instead. (And then it would have to flip on the previous one for bottom to top) |
| BlockTrapDoor | As vanilla doesn't doesn't care if a trap door doesn't face a particular side, we can probably do a similar thing to stairs. |
| BlockSkull | This is a bit trickier as the rotation data is either in the tile entity or in the block state. However it should be possible to do something with the skull - although perhaps treat the rotation types separately - never go from a wall-based skull to a ground-based one. |
| BlockCommandBlock | This might be useful, I'm not sure. This one is a bit sensitive due to command blocks being unforgiving to minor changes (it's basically code). |